content/post/quic/article.md

```diff
@@ -0,0 +1,76 @@
+
+## Connection Migration
+
+When a mobile device switches between WiFi and mobile data networks, existing TCP connections time out and the device has to reinitiate a new connection. This is caused by the way the kernel handles TCP connections: When receiving a packet, it looks at the 4-tuple (source and destination IP address and port) in order to associate it with an existing connection. This model worked well in the early days of the internet, when computer were stationary and IP addresses were stable. 
+
+For mobile devices switching between WiFi networks and mobile networks, this leads to interruptions every time the network changes. QUIC solves this problem by not relying on the 4-tuple. Connections are identified by so called connection IDs instead, which is contained in the header of every QUIC packet. To prevent linkability between the old and the migrated connections, peers will switch to a new connection ID they agreed upon before when switching between network interfaces.
+
+## Advanced Loss Recovery and Congestion Control
+
+Since QUIC uses UDP, it has to deal with recovering from a packet loss. QUIC's loss recovery algorithms build on decades of experience with TCP's loss recovery, and fixes some of TCP's problem.
+
+One source of uncertainty in TCP stems from the way packets are retransmitted. When TCP detects that a packet was (potentially) lost, it retransmits the packet as a whole, using the same sequence number. When receiving an acknowledgement for that sequence number, the sender can't be sure if this acknowledgement was generated when receiving the retransmission, or if the original packet was delayed.
+
+QUIC therefore never retransmits packets. When QUIC detects a packet as lost, it takes all the frames from that packet that need to be retransmitted, and sends a new packet as a retransmission. This retransmission gets a new packet number, so there will never be an ambiguity when receiving acknowledgements.
+
+## Encryption
+
+There's no such thing as an unencrypted QUIC connection. QUIC uses TLS 1.3 to establish key material to encrypt the connection. QUIC doesn't only encrypt all the application data exchanged over a QUIC connection, but also all internal information exchanged (e.g. acknowledgements for received packets, flow control updates, etc.). For an external observer, there's very little that can be learnt by looking at a QUIC flow: Except for the first packet containing the ClientHello TLS message (which necessarily can't be encrypted, since there's no shared key material between the endpoints yet), only the connection ID is unencrypted. Even the packet number is covered by QUIC's encryption.
```

Comment:
light grammer suggestions, no need to accept this if you don't think it helps 😄. Americans will want "learned" over "learnt".

```suggestion
There's no such thing as an unencrypted QUIC connection. QUIC uses TLS 1.3 to establish key material to encrypt the connection. QUIC not only encrypts all application data exchanged over a QUIC connection, but also all internal information exchanged (e.g. acknowledgements for received packets, flow control updates, etc.). For an external observer, very little that can be learnt by looking at a QUIC flow: Except for the first packet containing the ClientHello TLS message (which necessarily can't be encrypted, since there's no shared key material between the endpoints yet), only the connection ID is unencrypted. Even the packet number is covered by QUIC's encryption.
```